File: 367-valid-perfect-square/367-valid-perfect-square.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPerfectSquare = function(num) {
    var s = 1;
        var e = num;
        
        while(s<=e){
            const mid = Math.floor(s+(e-s)/2);
            
            var t = mid*mid;
            
            if(t==num){
                return true;
            }else if(t>num){
                e=mid-1;
            }else{
                s=mid+1;
            }
        }
        return false;
   
    
};

File: 367-valid-perfect-square/367-valid-perfect-square.java
class Solution {
    public boolean isPerfectSquare(int num) {
   
        int temp = num;
        
        for(int i=1;temp>0;i+=2){
            temp -= i;
        }
       if(temp ==0){
           return true;
       } else{
           return false;
       }
        
    }
}